### PR TITLE
Updates for HybridHadronization

### PR DESCRIPTION
**Pull request by mkordell**
This includes updates and bugfixes for HybridHadronization:

The main purpose of this update is a new color structure version of parton recombination, it both uses color information to determine recombination features and recombination will affect the color structure used for string fragmentation.

### CODE REVIEW

File: config/jetscape_master.xml [hunk 1]
```diff
@@ -261,11 +272,11 @@
    <JetHadronization>
        <name>colored/colorless/hybrid</name>
        <take_recoil>1</take_recoil>
-       <eCMforHadronization>2510</eCMforHadronization>
-       <reco_Elevelmax>3</reco_Elevelmax>
+       <eCMforHadronization>460</eCMforHadronization>
```
Comment (jdmulligan) [inline config/jetscape_master.xml:275]: Do these need to be updated? If there is a good reason to do it we can, however if not, we should keep them fixed (and it will break the automated tests)

File: config/jetscape_user.xml [hunk 2]
```diff
@@ -13,11 +13,22 @@
   
   <!-- Hard Process -->
   <Hard>
-    <PythiaGun>
```
Comment (jdmulligan) [inline config/jetscape_user.xml:16]: Why is this removed?

File: config/jetscape_user_twostagehydro.xml [hunk 3]
```diff
@@ -35,12 +37,51 @@
   </Hydro>
   
   <!-- Create liquifier -->
-  <Liquefier> </Liquefier>
+  <Liquefier>
```
Comment (jdmulligan) [inline config/jetscape_user_twostagehydro.xml:40]: Are these changes to the liquefier intended?

File: examples/unittests/causal_liquifier.cc [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * Copyright (c) The JETSCAPE Collaboration, 2018
  *
  * Modular, task-based framework for simulating all aspects of heavy-ion collisions
- * 
+ *
```
Comment (jdmulligan) [inline examples/unittests/causal_liquifier.cc:5]: Are these changes related to your intended commits? If not they may be an artifact of migrating your branch from JETSCAPE-COMP. You may want to make a new branch from scratch, starting from JETSCAPE/release-candidate (or equivalently, JETSCAPE/master) and add your changes on top.

**Review by jdmulligan — CHANGES_REQUESTED** [review]
Thanks @mkordell. It looks to me however that there is a bunch of additional stuff in this PR beyond the hybrid hadronization changes -- presumably picked up when migrating from JETSCAPE-COMP to JETSCAPE. Can you have a look, and if needed create a fresh branch starting from JETSCAPE/master and put your changes on top?

**Comment by mkordell** [comment]
Hi @jdmulligan - thanks so much for looking it over!  I did not intend for there to be any changes to the liquefier or related - I think I'll have to create a new branch from JETSCAPE/master and modify that instead as you recommended.

I can comment on some of the other changes in the code that I made:

     The changes to PGun, PythiaGun, and Matter (and xml) were intended to remove the "usehybridhad" flag.  This was implemented in the past to prevent colored hadronization for running over events where the initial color tags were set from the initial state module of choice, as it would crash.  Colored hadronization has been 'fixed' to handle this (not in this merge, but from earlier) but the flag had been left.

     The changes to brick.cc & .h while strictly speaking is not HH, it allows for HH to run in a finite length brick (ex. LBT would run an infinite length brick without this change).  No changes to the xml were needed as it reads from matter's "brickL" setting.  The addition of thermalpartonsampler.cc & .h are also related to running in a brick, but it's tied into the HH code (basically HH code that's in a separate file for organization).

     The addition of epemGun.cc & .h allows for running e+e- events (there were also some xml additions related to this).  I can leave this out if desired as I'm not sure that it's necessary for a public release, but I can keep it in if others want to be able to run e+e- events with the framework.

     Also, as for the alteration of the hadronization xml tags, the defaults for <reco_Elevelmax>, <sh_recofactor>, and <th_recofactor> do need to be changed - the physical meaning of the latter 2 within the code have been changed and the old defaults don't make much sense.  The old values for all three are valid numerically, but not likely physically proper.  Would leaving the old default values in the user xml allow for the automated tests to pass?

Thanks again for your time!  I really appreciate it!  I'll try and have the new branch ready for another look over here tonight - should I close out this PR?

**Comment by jdmulligan** [comment]
Hi @mkordell 

Great - all of the above sounds good.

Could you actually make two separate pull requests, to make life easier for the automated tests?

- (1) All of your changes **except** the modifications to the JetHadronization parameters in `jetscape_master.xml`
- (2) The changes to the JetHadronization parameters in `jetscape_master.xml`

That way we can separate the "breaking" from "non-breaking" changes (if I understand correctly), and can validate the bulk of the changes with the automated tests.

You can either close this PR, or you can "force push" to the HH_colupd branch and the PR here will automatically update.

Thanks!
